numpy/lib/function_base.py

```diff
@@ -1897,24 +1899,25 @@ def diff(a, n=1, axis=-1):
     if n < 0:
         raise ValueError(
             "order must be non-negative but got " + repr(n))
+
     a = asanyarray(a)
     nd = a.ndim
-    slice1 = [slice(None)]*nd
-    slice2 = [slice(None)]*nd
-    slice1[axis] = slice(1, None)
-    slice2[axis] = slice(None, -1)
-    slice1 = tuple(slice1)
-    slice2 = tuple(slice2)
+    rng = range(nd)
+    axis = rng[axis]
```

Comment:
Thank you for that. I was looking for something like that here: https://stackoverflow.com/q/45364139/2988730. If you post an answer, I will definitely accept it in preference to the `range`-based solution.